### ISSUE TITLE
Postgres extension segfault with a timestamp column

### ISSUE THREAD

**Issue by samstokes** [confluentinc/bottledwater-pg#37]
I can reproducibly crash the BW extension (and thus the Postgres backend) with the following SQL:

```sql
CREATE TABLE stamp (created_at TIMESTAMP NOT NULL);
-- CREATE TABLE
INSERT INTO stamp VALUES (NOW());
-- INSERT 0 1
```

If the Bottled Water client is already running (i.e. already connected to Postgres), the server crashes right as the INSERT completes.  That causes the BW client to exit also ("Could not receive replication data: server closed the connection unexpectedly").  If I then restart the BW client, the server segfaults again.  This suggests the crash happens when the extension is streaming updates via the replication protocol.

(This report is attempting to provide more detail on the crash mentioned in [my comment on #18](https://github.com/confluentinc/bottledwater-pg/issues/18#issuecomment-165270922).)

To reproduce:

 1. Run BW setup as in README
 2. Run the SQL above.

I have reproduced this with the latest Docker images (bottledwater:0.1 sha256:73cb533483f6978cacc62575854bebce2e66f535fffa991e48e988ceeac3018a, postgres-bw:0.1 sha256:c3c383b0240ff773f5839f967f57aa223df989e0dc1340d9749aade401374830), which as per [@ept's comment on #18](https://github.com/confluentinc/bottledwater-pg/issues/18#issuecomment-168073335) were built from 63d4d0e, latest master at time of writing.

I *can't* reproduce this (i.e. no crash, works as expected) with Docker images built from f5e5d83 (the commit to fix #18).  I also can't reproduce (no crash) with a schema containing only int columns:

```sql
CREATE TABLE things (id SERIAL PRIMARY KEY, thing INTEGER NOT NULL);
-- CREATE TABLE
INSERT INTO things (thing) VALUES (42);
-- INSERT 0 1
```

This makes me think there is a bad interaction between the changes to timestamp schema generation and the subsequent commits on master.

The server log is not especially informative, just stating that it segfaulted:

```
LOG:  server process (PID 91) was terminated by signal 11: Segmentation fault
LOG:  terminating any other active server processes
WARNING:  terminating connection because of crash of another server process
DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
HINT:  In a moment you should be able to reconnect to the database and repeat your command.
WARNING:  terminating connection because of crash of another server process
DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
HINT:  In a moment you should be able to reconnect to the database and repeat your command.
LOG:  all server processes terminated; reinitializing
```

**Comment by ept** [comment]
Thanks for the report, I was able to reproduce it and I think I've fixed it in bf6db77. The problem wasn't to do with timestamps, but with a table that had no primary key. Bottled Water really needs some automated tests... :grimacing:

**Comment by samstokes** [comment]
@ept sorry for the long pause, I can confirm that after bf6db77 I can no longer reproduce the issue.  Thanks for the fix!
